sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -188,6 +199,10 @@ class SessionCatalog(
 
   def setCurrentDatabase(db: String): Unit = {
     val dbName = formatDatabaseName(db)
+    if (dbName == globalTempDB) {
```

Comment:
When `globalTempDB` is set to a name that is not in the lower case, this compare is not right. Thus, `formatDatabaseName` need to be applied to both sides.